src/Services/Capsules/HasCapsules.php

```diff
@@ -21,7 +22,21 @@ public function getCapsuleList()
         $list = collect(config('twill.capsules.list'));
 
         if (!config('twill.capsules.loaded')) {
+            $composerCapsules = collect(InstalledVersions::getInstalledPackagesByType('twill-capsule'))
+                ->unique()
+                ->map(function ($fullName): array {
+                    [$vendorName, $packageName] = explode('/', $fullName, 2);
+
+                    return [
+                        'name' => $packageName,
+                        'fullName' => $fullName,
+                        'enabled' => true,
```

Comment:
I personally like the auto-discovery feature here! Does this mean that a composer capsule can't be disabled manually (or programmatically) via configuration? Curious to know your thoughts on this.

Reply:
Yeah, that's right, they can't currently be disabled. I couldn't think why it would be useful to be able to disable a Capsule when you can just composer remove it.

What is the use-case for disabling capsules?

Reply:
Good point, `composer remove` is probably the most straightforward way to remove a capsule.

Some use case I can think of for disabling capsules via configuration: debugging, customizing which capsules are available for a given user or context (e.g in a multi-tenant application). There are probably more use cases to uncover. I guess the main point I'm thinking of is: should we provide a way to opt-out of capsule auto-discovery? Similar to what is possible with Laravel packages.

Reply:
I think I agree. How about this, make autodiscovery optional (but enabled by default) and when it is enabled Composer installed capsules can not be disabled. This I believe will cover most use-cases.

Then if more control is needed autodiscovery can be disabled and then the developer is given full control over Composer capsules using the existing Capsule config, eg:

```
    'capsules' => [
        'auto_discover' => false, 
        'list' => [
            [
                'name' => 'test-twill-capsule',
                'composer' => true,
                'enabled' => false,
            ],
        ],
    ],
```

Reply:
Perhaps we can reverse the process and introduce a "TwillServiceProvider" that can be published.

In there during the boot/register (honestly I don't know which has what purpose). We can do:

$this->registerCapsule(ExampleCapsuleServiceProvider::class);

Reply:
This is also much easier to manage as a developer as opposed to registering capsules via config.

Reply:
@haringsrob See the later replies below. We looking more at using the inbuilt Laravel auto discovery to register capsules automatically when they are installed:

> ... I propose that instead of creating our own custom capsule package type we make use of the Laravels standard for installing and discovering packages. We could make use of a service provider to then register a capsule...